### DISCUSSION TITLE
Automate Folders

### DISCUSSION THREAD

**Discussion by fountainofkittens** [chhoumann/quickadd#552] [Q&A]
I want to understand how to automate folder creation.
My use case: I am documenting each plugin I install in Obsidian: settings, setup, functions, etc. 

- I want to create a series of nested folders with a templated folder note, with a path such as: `{{VALUE:Title}} Plugin/Settings - {{VALUE:Title}}/Settings List - {{VALUE:Title}}/` 

I am using MAKE.md for folder notes.

**Comment by chhoumann** [comment]
Answering this very late - sorry. Two building blocks get you exactly this:

1. Folder paths and file name formats in a Template choice accept the full format syntax, so `{{VALUE:Title}} Plugin/Settings - {{VALUE:Title}}` is a valid folder path - and QuickAdd creates any missing folders along the way automatically.
2. One Template choice creates one note, so for a folder note at each of your three levels you want three Template choices run together as a Macro. Named variables are shared within a single run, so `{{VALUE:Title}}` prompts once and all three choices reuse the answer.

Setup, using your structure:

- Template choice 1 - folder: `{{VALUE:Title}} Plugin`, file name format: `{{VALUE:Title}} Plugin`
- Template choice 2 - folder: `{{VALUE:Title}} Plugin/Settings - {{VALUE:Title}}`, file name format: `Settings - {{VALUE:Title}}`
- Template choice 3 - folder: `{{VALUE:Title}} Plugin/Settings - {{VALUE:Title}}/Settings List - {{VALUE:Title}}`, file name format: `Settings List - {{VALUE:Title}}`
- A Macro choice that runs the three in order (point each at your MAKE.md folder-note template).

With each file named the same as its folder, MAKE.md picks them up as folder notes. One prompt for the plugin title, and the whole tree plus three templated notes appear in one go.

Docs on format syntax in folder paths: https://quickadd.obsidian.guide/docs/Choices/TemplateChoice
